### ISSUE TITLE
Permission status is notAgain even on first lauch

### ISSUE THREAD

**Issue by harshapulikollu** [once10301/permission#19]
I'm testing the permissions on new device but also my permission status went into` notAgain`. It should be `allow` such that I can ask permissions from user. How to cope up with that.! What's the reason that cause permission status to be `notAgain`

**Comment by once10301** [comment]
When the user chooses not again, you should prompt the user Permission.openSettings.

**Comment by harshapulikollu** [comment]
It even not showing permission dialog for user to press not again or allow or deny, just on first launch logger prints that permission already went into not again state.

**Comment by b-cancel** [comment]
please @once10301 fix this, im having the same issue... for me it ONLY happen in RELEASE mode

**Comment by b-cancel** [comment]
BTW @once10301 in my case I can easily detect notAgain and then open app settings... HOWEVER... in my app settings the permission I requested does not show up as an option for the user to turn on... eventhough I have the permission in all of my adroidmanifest.xml files... both in the debug and the release version... Help would be very much appreciated since this has stopped my progression for days and frankly i can't find any way out except simply writing the plugin myself

**Comment by once10301** [comment]
@harshapulikollu On first lauch you should requestPermissions and then handle the latter things based on the returned results.

**Comment by harshapulikollu** [comment]
Yeah , I know that. But my question is **how come permission status is NotAgain or NoAgain even when user haven't denied requested permissions?**

**Comment by once10301** [comment]
if you do not have requests, ContextCompat.checkSelfPermission(registrar.activity(), permission) == PackageManager.PERMISSION_DENIED && ActivityCompat.shouldShowRequestPermissionRationale(activity, permission) == false, this is the same as  NotAgain

**Comment by b-cancel** [comment]
@once10301 thanks for the explanation

Although it would be nice to get this in the documentation
or perhaps edit your code a bit so that whoever is using your plugin does not have to worry about this detail

**Comment by lifenautjoe** [comment]
Had the same issue. Some docs with this would be greatly appreciated.

**Comment by iamsureshsharma** [comment]
Have the same issue @once10301, permission status is `notAgain` by default, is this issue fixed in `0.1.7`?
